kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -355,24 +356,60 @@ showLeafs
     => ReplM claim ()
 showLeafs = do
     leafsByType <- sortLeafsByType <$> getInnerGraph
-    case foldMap showPair leafsByType of
+    case Map.foldMapWithKey showPair leafsByType of
         "" -> putStrLn' "No leafs found, proof is complete."
         xs -> putStrLn' xs
   where
-    sortLeafsByType :: InnerGraph -> [(NodeState, [Graph.Node])]
-    sortLeafsByType graph =
-        groupSort
-            . catMaybes
-            . fmap (getNodeState graph)
-            . findLeafNodes
-            $ graph
+    showPair :: NodeState -> [Graph.Node] -> String
+    showPair ns xs = show ns <> ": " <> show xs
 
-    findLeafNodes :: InnerGraph -> [Graph.Node]
-    findLeafNodes graph =
-        filter ((==) 0 . Graph.outdeg graph) $ Graph.nodes graph
-
-    showPair :: (NodeState, [Graph.Node]) -> String
-    showPair (ns, xs) = show ns <> ": " <> show xs
+proofStatus
+    :: forall claim
+    . Claim claim
+    => ReplM claim ()
+proofStatus = do
+    graphs <- Lens.use lensGraphs
+    claims <- Lens.use lensClaims
+    let cindexes = ClaimIndex <$> [0..length claims - 1]
+    let allProofs = Map.union
+                        (inProgressProofs graphs)
+                        (notStartedProofs graphs cindexes)
+    putStrLn' $ Map.foldrWithKey acc "Current proof status: " allProofs
+  where
+    acc :: ClaimIndex -> GraphProofStatus -> String -> String
+    acc key elm res =
+        res
+        <> "\n  claim "
+        <> (show . unClaimIndex) key
+        <> ": "
+        <> show elm
+    inProgressProofs
```

Comment:
Unless I'm misunderstanding your comment here, I think I still need to keep the ClaimIndex for each GraphProofStatus so that's why I think it's simpler to keep this a function from Map to Map.

Reply:
Nevermind, you're right. Completely misunderstood what you said. Fixed this as well.